packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.ts

```diff
@@ -316,17 +342,23 @@ export function useGridColumns(
         return params;
```

Comment:
It will return `params` if `columnVisibilityModel` and `initialState.columns.columnVisibilityModel` is not present. That means that to use `exportState` users need to define these props, even if they are not going to use them. Is this a requirement to use this feature or a bug?

Reply:
The problem is not that we don't support restoring the columns when `hide` is used. It's not worth to support it since it's a legacy API. The point in my comment was that to able to restore the state with `restoreState` the user must set `columnVisibilityModel` to some value. Maybe we should improve the condition used to set `shouldUseVisibleColumnModel` to assume that we're using the new approach if no column is using `hide`. Currently, for it to work when I've no intention of controlling the column visibility I need to pass `columnVisibilityModel={}`.

Reply:
Oh right I forgot to set `shouldRegenColumnVisibilityModelFromColumns` to `true` when using the legacy method